app/scripts/proactive/view/dom.js

```diff
@@ -149,14 +149,27 @@ define(
             }
         });
 
-        $("#set-templates-bucket-button").click(function (event) {
+        $("#set-templates-main-bucket-button").click(function (event) {
             event.preventDefault();
             var studioApp = require('StudioApp');
             if (studioApp.isWorkflowOpen()){
                 studioApp.models.catalogBuckets.fetch({reset: true, async: false});
                 studioApp.modelsToRemove = [];
-                studioApp.views.catalogSetTemplatesBucketView.render();
-                $('#set-templates-bucket-modal').modal();
+                studioApp.views.catalogSetMainTemplatesBucketView.render();
+                $('#set-templates-main-bucket-modal').modal();
+            }else{
+                $('#open-a-workflow-modal').modal();
+            }
+        });
+
+        $("#set-templates-secondary-bucket-button").click(function (event) {
```

Comment:
this function is the same as the previous one except that one is main and this is secondary. Do you think we can have a single external generic function that accepts a parameter to do one or the other?